src/main.ts

```diff
@@ -1,101 +1,111 @@
-            core.error(`Azure PowerShell Login failed. Please check the credentials. For more information refer https://aka.ms/create-secrets-for-GitHub-workflows"`);
-        }
-        core.setFailed(error);
-    } finally {
-        // Reset AZURE_HTTP_USER_AGENT
-        core.exportVariable('AZURE_HTTP_USER_AGENT', prefix);
-        core.exportVariable('AZUREPS_HOST_ENVIRONMENT', azPSHostEnv);
-    }
-}
-
-async function executeAzCliCommand(command: string, silent?: boolean) {
-    try {
-        await exec.exec(`"${azPath}" ${command}`, [],  {silent: !!silent}); 
-    }
-    catch(error) {
-        throw new Error(error);
-    }
-}
-
+import * as core from '@actions/core';
+import * as crypto from "crypto";
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+import { FormatType, SecretParser } from 'actions-secret-parser';
+import { ServicePrincipalLogin } from './PowerShell/ServicePrincipalLogin';
+import { AzureCloudName} from './PowerShell/Enums';
+
+var azPath: string;
+var prefix = !!process.env.AZURE_HTTP_USER_AGENT ? `${process.env.AZURE_HTTP_USER_AGENT}` : "";
+var azPSHostEnv = !!process.env.AZUREPS_HOST_ENVIRONMENT ? `${process.env.AZUREPS_HOST_ENVIRONMENT}` : "";
+
+async function main() {
+    try {
+        // Set user agent variable
+        var isAzCLISuccess = false;
+        let usrAgentRepo = crypto.createHash('sha256').update(`${process.env.GITHUB_REPOSITORY}`).digest('hex');
+        let actionName = 'AzureLogin';
+        let userAgentString = (!!prefix ? `${prefix}+` : '') + `GITHUBACTIONS/${actionName}@v1_${usrAgentRepo}`;
+        let azurePSHostEnv = (!!azPSHostEnv ? `${azPSHostEnv}+` : '') + `GITHUBACTIONS/${actionName}@v1_${usrAgentRepo}`;
+        core.exportVariable('AZURE_HTTP_USER_AGENT', userAgentString);
+        core.exportVariable('AZUREPS_HOST_ENVIRONMENT', azurePSHostEnv);
+
+        azPath = await io.which("az", true);
+        await executeAzCliCommand("--version");
+
+        let creds = core.getInput('creds', { required: true });
+        let secrets = new SecretParser(creds, FormatType.JSON);
+        let servicePrincipalId = secrets.getSecret("$.clientId", false);
+        let servicePrincipalKey = secrets.getSecret("$.clientSecret", true);
+        let tenantId = secrets.getSecret("$.tenantId", false);
+        let subscriptionId = secrets.getSecret("$.subscriptionId", false);
+        let resourceManagerEndpointUrl = secrets.getSecret("$.resourceManagerEndpointUrl", false);
+        let environment = core.getInput("environment");
+        const enableAzPSSession = core.getInput('enable-AzPSSession').toLowerCase() === "true";
+        if (!servicePrincipalId || !servicePrincipalKey || !tenantId || !subscriptionId) {
+            throw new Error("Not all values are present in the creds object. Ensure clientId, clientSecret, tenantId and subscriptionId are supplied.");
+        }
+
+        // Attempting Az cli login
+        if (environment.toLowerCase() == "azurestack") {
```

Comment:
move this entire if block in a separate method like RegisterAzureStackEnvironment()

Reply:
@asranja since this is an existing code of Azure stack I am not very comfortable on touch stack code.And for that reason we branched out separately for Soverign cloud.